server/src/main/java/org/elasticsearch/index/mapper/IgnoredFieldMapper.java

```diff
@@ -59,14 +67,21 @@
         public static final IgnoredFieldType INSTANCE = new IgnoredFieldType();
 
         private IgnoredFieldType() {
-            super(NAME, true, false, TextSearchInfo.SIMPLE_MATCH_ONLY, Collections.emptyMap());
+            super(NAME, true, Version.CURRENT.id >= Version.V_7_1_0.id,
```

Comment:
Doc values should be activated only if the index is created in this version. You don't have the information available here so this check should move to the FixedTypeParser. The check for the version could look like this:
```
if (c.indexVersionCreated().onOrAfter(Version.V_8_0_0)) {
...
```
Note that I am using `Version.V_8_0_0` because the PR targets master. We'll adapt the version check **after** the backport to 7.x.